pytorch_lightning/loggers/mlflow.py

```diff
@@ -158,9 +157,11 @@ def log_metrics(self, metrics: Dict[str, float], step: Optional[int] = None) ->
 
             new_k = re.sub("[^a-zA-Z0-9_/. -]+", "", k)
             if k != new_k:
-                warnings.warn(("MLFlow only allows '_', '/', '.' and ' ' special characters in metric name.\n",
-                               f"Replacing {k} with {new_k}."))
-            k = new_k
+                rank_zero_warn(
+                    "MLFlow only allows '_', '/', '.' and ' ' special characters in metric name."
+                    f" Replacing {k} with {new_k}.", RuntimeWarning
```

Comment:
Just a note, I think we use UserWarning across Lightning.

Reply:
not sure what's the difference, but `RuntimeWarning` is used in many places in the codebase. You can do a simple grep search to check.